src/org/thoughtcrime/securesms/ConversationActivity.java

```diff
@@ -743,6 +758,49 @@ private void handleViewPinnedMessages() {
     startActivity(intent);
   }
 
+  private void handleSearch(MenuItem item) {
+    bottomPanel = (InputPanel) findViewById(R.id.bottom_panel);
+    searchView = (SearchView)  findViewById(R.id.custom_search);
+
+    searchView.setOnQueryTextListener(new SearchInitiatedListener());
+    upArrow.setOnClickListener(new NextSearchResultListener());
+    downArrow.setOnClickListener(new PreviousSearchResultListener());
+
+    if(!this.isSearchMode) {
+      showSearchMode(bottomPanel, searchView);
+    } else {
+      hideSearchMode();
+    }
+    this.isSearchMode = !isSearchMode;
+  }
+
+  public void hideSearchMode() {
+    hideKeyboard();
+    linSearch.setVisibility(View.GONE);
+    bottomPanel.setVisibility(View.VISIBLE);
+    linMessage.setVisibility(View.VISIBLE);
+    this.isSearchMode = !isSearchMode;
```

Comment:
i think it would be preferable to set this to a boolean since it should always be the same.